content/blog/how-to-registries/index.md

```diff
@@ -0,0 +1,210 @@
+---
+title: "How to Build a Container Registry"
+date: 2020-08-27
+draft: true
+meta_desc: "How to build a registry and push Docker images with code."
+meta_image: registry.png
+authors:
+    - sophia-parafina
+tags:
+    - containers
+    - registry
+    - Docker
+---
+
+Whether you are working with Kubernetes or serverless, your application uses containers. If you use the Docker desktop client, images are pushed to Docker Hub by default. Pulling images from Docker Hub is convenient, but there are many reasons to store images in your registry. For example, Docker Hub doesn’t guarantee to produce the same image on repeated pulls, i.e., your base image might have changed. It’s also possible to inadvertently expose secrets in an intermediate image used to build the image stored on Docker Hub. There is also the possibility of vulnerabilities in even official images. This article shows how to create a repository and how to build and push images to that repository
+
+<!--more-->
+
+## Container registry offerings
+
+The AWS provides the [Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/), Azure has  [Container Registry](https://azure.microsoft.com/en-us/services/container-registry/), and Google has it’s [Container Registry](https://cloud.google.com/container-registry). Each provider has associated services unique to their offering, but all support Docker or OCI compliant images.
+
+## Build it
+
+Let’s examine how to create a registry with the provider of your choice. In these examples, we create a registry, build a Docker image, and push the image to the registry. The [application](https://github.com/pulumi/examples/tree/master/aws-ts-containers/app) used for the image is NGINX.
+
+Choose your cloud provider to learn how to build a registry.
+
+{{< chooser cloud "aws,azure,gcp" >}}
+{{< choosable cloud aws >}}
+
+In this example, we create an ECR repository configured to scan an image’s Operating System components. Scanning for vulnerabilities in an application is currently [out of scope](https://aws.amazon.com/blogs/containers/amazon-ecr-native-container-image-scanning/). We also set a policy for the repository that controls the actions allowed and a lifecycle policy that expires an image after a set time.
+
+With Pulumi, it’s possible to build an image locally using Docker and push it to your repository. To push the image, we obtain the credentials required to push from the registry. Finally, we export the credentials and the URL for the registry. Read more about ECR in the [API Reference]({{< relref "/docs/reference/pkg/aws/ecr" >}}).
+
+```typescript
```

Comment:
Should we not have each language example?

Reply:
We should and I'll try to vary the language more in future blog posts.